Fix for file inconsistencies in file system and main editor

lognguyenle opened the pull request:
Attempt to fix #73 and #59

Overview:
- Select interactions result in file data being overwritten or shown incorrectly:
  - Double clicking on a file A in the editor, switching to file B, switching to file A results in file A now containing file B's data
  - Converting or adding new files/models then switching between files results in file content inconsistencies as well
  
Changes:
- In AntimonyEditor component:
  - make sure file is loaded from database with content before the rest of the component is rendered
  - Editor now changes the focused file on the state change of 'selectedFile'
  - Files save only after changes from user
  - On model search, files are added to the database, saved, then switched to in the editor
 
- In main App:
  - SBML and Antimony conversion were complicated, rely on convertAntimonytoSBML and convertSBMLtoAntimony methods instead of processSBML and processAntimony
    - removed sbmlResultHandler, antimonyResultHandler and associated states and lines
   
Tests:
- could not get AWE to build locally
- passes all working tests in `npm run test`

kevincnguyen commented on src/components/antimony-editor/AntimonyEditor.tsx:
```diff
@@ -313,24 +345,14 @@ const AntimonyEditor: React.FC<AntimonyEditorProps & { database: IDBPDatabase<My
           monaco.editor.defineTheme("antimonyTheme", antimonyTheme);
           monaco.editor.setTheme("antimonyTheme");
           // Retrieve content and file name from IndexedDB
```
nit: Try to remove any outdated comments or code you commented out to keep the codebase clean. It's often hard to catch though, so no worries!

kevincnguyen commented on src/components/antimony-editor/AntimonyEditor.tsx:
```diff
@@ -340,17 +362,15 @@ const AntimonyEditor: React.FC<AntimonyEditorProps & { database: IDBPDatabase<My
               wordWrapColumn: 80,  // Max column width before wrapping
               wrappingIndent: 'same',
             });
             window.antimonyActive = true;
```
nit: I think `window.antimonyActive` can be removed as it is not used for anything anymore.

kevincnguyen commented on src/App.tsx:
```diff
@@ -133,149 +128,59 @@ const App: React.FC = () => {
   const handleConversionAntimony = () => {
     try {
-      if (window.processAntimony) {
-        window.processAntimony();
```
To keep the codebase clean, I believe we can remove `processAntimony` and also `processSBML` can be removed from Conversion.js (both in the `root` and `public` folders) and any remaining mentions of them as both methods are not used anymore.

lognguyenle replied:
Thanks for the thorough review and feedback! I added the requested changes. Should I feel free to merge?

kevincnguyen reviewed (changes requested):
I manually tested and built the code and it all seems to work as intended! I also agree with you that the previous SBML and Antimony conversion was complicated, so I like the changes you made. 

Overall, it looks good to me, but I think you can clean up some lines of code before merging!

lognguyenle commented:
Changes:
- removed leftover and remnant codes and comments (all of them I think)
- removed window.antimonyActive global variable
- removed selectedFileContent and handleConversionSBML from AntimonyEditor props
- rewrote a couple of comments mentioning old methods
- added some null checking for ease of bug-fixing
- added a couple of new comments for certain lines

kevincnguyen reviewed (approved):
Looks good to me! I would wait until Edison reviews this before merging, as I could've missed something.

Edih112 reviewed (approved):
lGTM! Tested manually and the fix seems to work, and the codebase is more concise now which is a huge plus! 

Side note: for `npm run build` to work you need to make the changes to `path.js` described in https://github.com/sys-bio/AntimonyEditor/blob/master/DEV_INSTRUCTIONS/DevInstructions.md (these are just temporary and only needed on deployment/build.
